src/swarm.js

```diff
@@ -20,7 +20,7 @@ function Swarm () {
 
   self.port = parseInt(process.env.IPFS_SWARM_PORT, 10) || 4001
   self.connections = {} // {peerIdB58: {conn: <>, socket: <>}
-  self.handles = []
+  self.handles = {}
```

Comment:
Any specific reason on having handles as an object instead of an array? I see you changed the looped and both work fine. I might just be missing a perf improv, perhaps?